Build a regression model.

In [9]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import statsmodels.api as sm

df = pd.read_csv('Master_new.csv')
df_unique = df.drop_duplicates(subset=['name_y', 'location.display_address'], keep='first')
df_unique.dropna(inplace=True)


/var/folders/fz/ntgp3y9j3p777vzmhx_1fmwr0000gn/T/ipykernel_6267/2463674171.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_unique.dropna(inplace=True)


Provide model output and an interpretation of the results. 

In [8]:
y = df_unique['distance_y']
x= df_unique[['review_count','rating']]

x = sm.add_constant(x)
results = sm.OLS(y,x).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             distance_y   R-squared:                       0.004
Model:                            OLS   Adj. R-squared:                 -0.011
Method:                 Least Squares   F-statistic:                    0.2564
Date:                Tue, 08 Nov 2022   Prob (F-statistic):              0.774
Time:                        01:34:27   Log-Likelihood:                -985.63
No. Observations:                 138   AIC:                             1977.
Df Residuals:                     135   BIC:                             1986.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const          750.4299    179.772      4.174      0.000     394.896    1105.964
review_count     0.0501      0.073      0.686      0.494      -0.094       0.194
rating          -0.4487     49.443     -0.009      0.993     -98.232      97.335
==============================================================================
Omnibus:                        8.119   Durbin-Watson:                   1.356
Prob(Omnibus):                  0.017   Jarque-Bera (JB):                7.790
Skew:                          -0.526   Prob(JB):                       0.0203
Kurtosis:                       2.503   Cond. No.                     3.59e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.59e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

R-squared reflects the fit of the model. R-squared values range from 0 to 1, where a higher value generally indicates a better fit, assuming certain conditions are met. 
In this output, we can see that the value is only 0.0004. This means that the model is not capable of explaining the patterns in the data.


A p-value of less than 0.05 is considered to be statistically significant. 
This regression output shows a p-value for review-count is 0.0494 and rating is 0.993,which means that the probability of the relationship between the ditance does NOT  impact the review counts and ratings 

# Stretch

How can you turn the regression model into a classification model?

I didnt have time for this stretch question right now. will come back later to think about it.

Thank you! 